### DISCUSSION TITLE
SVG loading

### DISCUSSION THREAD

**Discussion by Edgiest05** [raysan5/raylib#3741] [Features]
I've read in the latest release about the introduction of "SVG loading" and while attempting to use them I've noticed that its support is still disabled in [config.h](https://github.com/raysan5/raylib/blob/master/src/config.h#L164).
Is such functionality still a work in progress or simply opt-in?

**Comment by raysan5** [comment]
> Is such functionality still a work in progress or simply opt-in?

It's simply opt-in, most users do not need it and including it (like many other file-formats) implies additional code and executable size.

**Comment by clankill3r** [comment]
How do you enable? In config.h the word svg is not even present.

**Reply by Edgiest05** [reply]
It is being moved to `raylib-extras`. Commit 1effe92 shows config.h and related files being changed.
Sadly I'm not working with this atm and I can't provide a solution to how to enable it with the latest changes.

**Reply by JulienLecoq** [reply]
I'm also looking on how to do it.

**Reply by raysan5** [reply]
It has been added to `raylib-extras`: https://github.com/raylib-extras/examples-c/commit/1e38dd9f1ae84c3beafecf69a7002a9339d1b4e9

**Comment by raysan5** [comment]
https://github.com/raylib-extras/examples-c/commit/1e38dd9f1ae84c3beafecf69a7002a9339d1b4e9

**Reply by abid-sayyad** [reply]
Hey @raysan5 do you have any example in loading the SVG as animation. This examples shows the implementation by static rendering the SVG. Can you help me out with loading animated SVG for example the one I am attaching.
![Glowing Line@1x-3 2s-1280px-200px](https://github.com/user-attachments/assets/9e5c4649-64b7-4d0d-89ac-871103c3b704)

**Reply by raysan5** [reply]
I'm afraid it is not supported by the library used but I actually don't know... I recommend using a spritesheet or loading a gif as frames

**Reply by abid-sayyad** [reply]
I wish to screate a GUI, which ahs animations on press(touch) events. What would be a better approach for creating this GUI.

1) using GIF with clear backgrounds to render
2) using video to display animations? (I was trying the example from Raylib that uses pl_mpeg, but I get an error stating "UNCOMPRESSED_R8G8B8" was not declared)
3) having a collection of image and rendering them one by one.

**Reply by abid-sayyad** [reply]
I require shorter animations, mostly playing during wait duration when my major processes are running in background. The waiting animation loads and plays on repeat. There is this that I will need to play two/three animations playing in the system idle state. Also all these things are running on a MPU. 

Video decoding as you mentioned is a bit more complex yes, I was looking at using pl_mpeg and/or vlc for extracting frames and displaying renders using raylib. It requires me to cross-compile the library and upload it to the device. It is then only I can work on the application.

**Reply by abid-sayyad** [reply]
I ended up using GIFs for the application, did the job.
Thanks for all the support and replies on the query.
